beamline.rst

```diff
@@ -1,212 +1,379 @@
 
 
     def get_actuator(name) -> ActuatorData:
-    """
-    :returns: The ActuatorData object identified by the given name
-    :rtype: ActuatorData
-    """
-       pass
+        """
+        :returns: The ActuatorData object identified by the given name
+        :rtype: ActuatorData
+        """
+        pass
 
 
     def set_actuator_value(name, value:Any) -> bool:
-    """
-    Tries to move the actuator identified by name to value
-
-    (Errors and progress of movement is passed asynchronously
-     via the available signaling mechanism)
-    
-    :returns: True if motion was started False otherwise
-    """
+        """
+        Tries to set the actuator identified by name to value.
+        Setting a disallowed value will raise ValueError, with one exception:
+        if the actuator takes a float value and has a non-empty values_list,
+        the value will be set to the nearest value in the list.
+
+        (Errors and progress of movement is passed asynchronously
+         via the available signalling mechanism)
+
+        :returns: True if motion was started False otherwise
+        """
         pass
 
+  **State/Value enumerations**
+
+      class ActuatorState(Enum):
+
+        # This enumeration should be limited to what the UI needs to know,
+        # not what the motors might want to tell. These values may need fixing.
+        # Why do we need MOVESTARTED, for instance?
+
+        NOTINITIALIZED = 0  # Actuator has not yet been set up. value is None
+        UNUSABLE = 1        # Actuator is not functional. value is None
+        READY = 2           # Actuator is functional and ready to accept new moves.
+        MOVESTARTED = 3     # Move has started. Why do we need this?
+        MOVING = 4          # Actuator is moving and does not accept move orders.
+                            # Value is defined but unstable.
+        ONLIMIT = 5         # Value is on limit. Actuator accepts move orders (?)
+        FROZEN = 6          # Actuator is functional, but cannnot be moved.
+                            # value is defined, and may be modified by HO level.
+                            # Needed for e.g. wavelength on non-tunable beamlines,
+                            # machine_current, fill_mode.
+
+      class TwoStateValue(Enum):
+
+        # There are two states, with aliases, the ACTIVE/IN/CLOSED state
+        # and the INACTIVE/OUT/OPEN
+        # As a mnemonic, you could say that 1 is for when the object is
+        # 'doing its job' (shutter closed, beamstop and frontlight in, ...)
+        # That means that for collection you need beamstop IN, and frontlight OUT
+        #
+        # NB Do we need an (oxymoronic) third state, like UNUSABLE?
+
+        INACTIVE = 0
+        OUT = 0
+        OPEN = 0
+
+        ACTIVE = 1
+        IN = 1
+        CLOSED = 1
+
+
+Specific Actuators:
+-------------------
+
+**Type float:**
+
+    *Centring motors*
+
+        Omega, Kappa, Phi, AlignmentX,  AlignmentY,  AlignmentZ,
```

Comment:
So these are actuator names ?

Reply:
That was the proposal - in order to get a single, high-level set of names that did not depend on local variations. I am fully prepared for people to knock that one down (though I did thake the Alignment and Centring from Martin Savko), but I thought it would be useful to start a discussion at least.